### ISSUE TITLE
Suggestion: Offer a F1 or F3 menu option for /unstuck

### ISSUE THREAD

**Issue by NEURALDELVERM** [tsunamayo/Starship-EVO#5189]
### Short Description of the issue:
   When two child entities move together they have a high likelihood of clipping into each other, even if that collision is only one point. Then, if the child entities move quickly enough, they can jam together. Now, I am not advocating for all-the-time clipping, but at least in creative mode, we should be able to "reset all child entity positions" or something like that. 
   Or alternatively, a dialog box that will pop up automatically after a certain number of collisions are detected, and the entities do not move for a period of time, like the player is intending to. The dialog box would say "Feeling a bit stuck? Click here to toggle child entity collision until the logic has completed and it's returned to resting state."
   While I can see cases of abuse using this, I can also see it being useful for creative building. In the future, when the Ship Yard concept has been released, maybe it would be nice to be able to turn child entities into "holograms" so that we can set them up for success before they become collide-able physics objects.
   I hope this is a useful QoL tool for builders, at least until there can be some changes in collision checks between moving child entities. I feel like speed of movement plays a huge factor because the movements can skip collision check ticks, then once stuck cannot move quickly enough to be fixed without cutting and pasting it back in after moving the conflicting entity. I know it's fixable but while not having undo, you can cause great harm to a craft design in the wrong circumstances. And if self-collision can cause damage in the future that could be catastrophic.
...

### Steps to reproduce the issue:
Set up a contraption with two child entities, and have them move at the same time. If they collide during this movement, based on their speed and the rotation of the collision point vectors, they can clip into each other. High speeds of movement between for example, pistons and hinges, caused my canopy design to cease functioning entirely after a few sticky hang-ups went bad. I could have prevented it with more sound logic gates, but this is a bug regardless of my design.
...

### Environment
- Starship EVO version: 23w29d
- Graphics Card: RX 480 4GB

### Additional information:
[NDM - DxDiag - JUL-22-23.txt](https://github.com/tsunamayo/Starship-EVO/files/12138113/NDM.-.DxDiag.-.JUL-22-23.txt)
_(Screenshots, Videos, Error logs, Related issues)_
![image](https://github.com/tsunamayo/Starship-EVO/assets/23009112/e1547593-fefe-4f61-9347-971e329757a2)
![711980_20230722222924_1](https://github.com/tsunamayo/Starship-EVO/assets/23009112/cd901506-e915-4f56-bf01-e5a6cf65bbb3)

**Comment by NEURALDELVERM** [comment]
I can post the craft if you need this example, it could be related to certain blocks interactions and collision bounds.

**Comment by Dwarf-LordPangolin** [comment]
> Now, I am not advocating for all-the-time clipping, but at least in creative mode, we should be able to "reset all child entity positions" or something like that.

Having spent an appreciable portion of today trying to fix some turrets that did precisely what you describe, I **_heartily_** support a creative command to reset the position of child entities to their zero position to fix instances of undesired clipping.

To give an illustration of why this is a very badly needed tool in creative: I'm making a ship with recessed turrets. They work fine, with no clipping, when testing them by moving the rotators. But due to bug #4963 they go crazy when testing the turret operation against targets, and then they're stuck.

There is currently _no way_ to unstick them, short of manually deleting every block that is obstructing movement on either the turret or parent ship. This is hardly the only scenario where buggy clipping can cause issues, and without the kind of tool suggested in this post, if the player doesn't happen to have a blueprint that was saved before the clipping started (which might not be noticed right away), they may be forced to redo a significant amount of work. I got lucky today, but I came within an ace of having to do precisely that, and it would be nice to not need to worry about the Sword of Clipocles hanging over my head in the future.

A debug command or option to fix these issues in creative would be _extremely_ helpful.

**Comment by NEURALDELVERM** [comment]
Another possibility requiring more work, could be to show a hologram and then use that hologram's collisions to show what blocks and entities would conflict by highlighting them in a red striped pattern or something, that way you can try and prevent this ahead of it happening. Now, that would be more shader work, etc, but I don't think it would be super complex. It would be a ghost version of the original thing, that could replace the child entity while in "collision test mode" or "collision debug mode" for instance. Again, this idea is just a footnote.

**Comment by Majikmonster** [comment]
I too have suffered this problem since the game was called Skywanderers. No matter how simple the mechanism or easy the fix on the builder's side, having to deconstruct and repair part of a build due to CE clipping that can't be reversed is frustrating to say the least. I've had everything from a simple 1-hinge canopy to turrets to multi-CE mechanisms become physically stuck, their movement paused, or ghosting halfway into the ship and not be able to simply reverse the movement to unstick them. Noclip mode for players was one of the best builder QoL features added for a similar reason. It would be fantastic to extend it to entities for construction and testing purposes.

**Comment by Kaiser-Indrasil** [comment]
This. Building with tightly packed CEs positioned at different angles is such a pain because of that. Oftentimes, it looks like the bricks do not physically touch, only for you to find out that their collision boxes do, so you have to build a CE all over again. Such a pain in the ass this is.

**Comment by ProPeach** [comment]
Something like this could also help minimise damage done to bugs like #5083!

**Comment by CoasterBlock** [comment]
Agreed, this would be great. Collisions dont work well anyway. Most child-entities clip very easily but its difficult to unclip them. 
A simple "return to default position" button would be nice. The default position would be read from connected logic such as sequencers.

**Comment by tsunamayo** [comment]
@NEURALDELVERM @ProPeach @Dwarf-LordPangolin @Kaiser-Indrasil  Okay I will look what I can do!

**Comment by tsunamayo** [comment]
So I have been thinking about that, it is a bit hard to do in a way that is robust. You will always get into nasty case as soon as several CE gets involved. Could you guys send me an example of such a jammed mecanism? I would prefer working a solution to eradicate jamming:  movement inside a CE would still be allowed if reduce the penetration distance.

@Kaiser-Indrasil It is not perfect for sure, as I have constraints to make it viable in real time. Also it seems you like to use them super tightly packed to get to some specific shape, which is not really what they are intended for ;). But if you find something that look like a bug please report it!
